src/battle_controllers.c

```diff
@@ -1546,3 +1559,1484 @@ void BtlController_EmitDebugMenu(u8 bufferId)
+        break;
+    case REQUEST_HP_BATTLE:
+        SetMonData(&party[monId], MON_DATA_HP, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_MAX_HP_BATTLE:
+        SetMonData(&party[monId], MON_DATA_MAX_HP, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_ATK_BATTLE:
+        SetMonData(&party[monId], MON_DATA_ATK, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_DEF_BATTLE:
+        SetMonData(&party[monId], MON_DATA_DEF, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SPEED_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SPEED, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SPATK_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SPATK, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SPDEF_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SPDEF, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_COOL_BATTLE:
+        SetMonData(&party[monId], MON_DATA_COOL, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_BEAUTY_BATTLE:
+        SetMonData(&party[monId], MON_DATA_BEAUTY, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_CUTE_BATTLE:
+        SetMonData(&party[monId], MON_DATA_CUTE, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SMART_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SMART, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_TOUGH_BATTLE:
+        SetMonData(&party[monId], MON_DATA_TOUGH, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SHEEN_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SHEEN, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_COOL_RIBBON_BATTLE:
+        SetMonData(&party[monId], MON_DATA_COOL_RIBBON, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_BEAUTY_RIBBON_BATTLE:
+        SetMonData(&party[monId], MON_DATA_BEAUTY_RIBBON, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_CUTE_RIBBON_BATTLE:
+        SetMonData(&party[monId], MON_DATA_CUTE_RIBBON, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_SMART_RIBBON_BATTLE:
+        SetMonData(&party[monId], MON_DATA_SMART_RIBBON, &gBattleResources->bufferA[battler][3]);
+        break;
+    case REQUEST_TOUGH_RIBBON_BATTLE:
+        SetMonData(&party[monId], MON_DATA_TOUGH_RIBBON, &gBattleResources->bufferA[battler][3]);
+        break;
+    }
+
+    if (GetBattlerSide(battler) == B_SIDE_PLAYER)
+        HandleLowHpMusicChange(&party[gBattlerPartyIndexes[battler]], battler);
+}
+
+void StartSendOutAnim(u32 battler, bool32 dontClearSubstituteBit)
+{
+    u16 species;
+    u32 side = GetBattlerSide(battler);
+    struct Pokemon *party = GetBattlerParty(battler);
```

Comment:
Actually I think we can just consolidate all of these GetBattlerParty and gBattlerPartyIndexes into just GetBattlerPartyData calls. Sorry I should have noticed earlier